libraries/Microsoft.Bot.Builder/Slot.cs

```diff
@@ -10,15 +10,15 @@ namespace Microsoft.Bot.Builder
 {
     public class Slot : IReadWriteSlot
     {
-        public Slot(IFrame frame, ISlotDefinition definition)
+        public Slot(IFrame frame, SlotDefinition definition)
```

Comment:
This coupling with the `IFrame` will be super confusing for customers. See comments on `IFrame::AddSlot` for suggestions on how to change the API which would include making this entire class an internal implementation detail as they would only ever work with the `IReadWriteSlot` interface.